### PR TITLE
Redirect documentation overhaul

### PR DESCRIPTION
**Pull request by alexfornuto**
**Note:** Please fill out the PR Template to ensure proper processing.

Closes, but not limited to #5493 

## Effect
PR includes the following changes:
- Updates and improves style and content for redirect-based documentation.
- Additional warnings about conflicts between Primary Domain and custom PHP redirects
- Splits content from `/redirects/` into `/advanced-redirects/`.

Edit: as an aside, this also implements a markdownlint ruleset for partial files, to ignore warnings about not starting with an H1 in those files.

## Remaining Work
- [x] Add IP-based restriction examples
    - [x] Include link to Advanced CDN content if available
- [x] +1 from @rachelwhitton and/or @tmountjr on redirect warning when going from `olddomain.com` to `newdomain.com/subpage`
- [x] +1 from @rachelwhitton on access restriction examples
- [ ] Copy Review
- [x] Update landings yaml file for new page

## Post Launch
**Do not remove** - To be completed by the docs team upon merge:
- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (*applicable*)
- [ ] Include `/advanced-redirects` within docs search service provider (*applicable*)
- [ ] Update Status Report
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

File: source/partials/primary-domain.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 ### Set a Primary Domain via the Dashboard
 
+<Alert type="danger" title="Warning">
+
+With a Primary Domain set at the platform level, all other domains will be pointed to your Primary domain _at the root level_. If you want to redirect secondary domains to specific pages on your site (for example, `olddomain.com` to `newdomain.com/old-landing-page`), do not set a Primary Domain. Instead use [PHP redirects](/redirects/#redirect-with-php).
+
+</Alert>
+
```
Comment (alexfornuto) [inline source/partials/primary-domain.md]: @tmountjr @rachelwhitton This is the warning that resolved #5493

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5516 | Difference |
| --- | --- | --- | --- |
| Performance Score | `55.00000000000001` | `55.00000000000001` | `0` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5516` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18399/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18399/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5516 | Difference |
| --- | --- | --- | --- |
| Performance Score | `51` | `62` | `11` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5516` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18410/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18410/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

File: source/content/advanced-redirects.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+The following configuration will redirect requests for `subdomain.example.com` to `https://example.com/subdirectory/`.
+
+When using this type of redirect, you must first [remove the primary domain](/redirects#update-or-remove-primary-domain) from the environment.
+
+```php:title=wp-config.php%20or%20settings.php
+// Redirect subdomain to a specific path.
+// Check if Drupal or WordPress is running via command line
+if (isset($_ENV['PANTHEON_ENVIRONMENT']) && ($_SERVER['HTTP_HOST'] == 'subdomain.example.com') && (php_sapi_name() != "cli")) {
+  $newurl = 'https://www.example.com/subdirectory/'. $_SERVER['REQUEST_URI'];
```
Comment (tmountjr) [inline source/content/advanced-redirects.md]: I _think_ this should be `$newurl = 'https://www.example.com/subdirectory' . $_SERVER['REQUEST_URI'];`

`$_SERER['REQUEST_URI']` should include the leading `/`.

File: source/partials/primary-domain.md (hunk 1 above)
Comment (tmountjr) [reply source/partials/primary-domain.md]: This is good!

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5516 | Difference |
| --- | --- | --- | --- |
| Performance Score | `56.00000000000001` | `59` | `3` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5516` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18415/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18415/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5516 | Difference |
| --- | --- | --- | --- |
| Performance Score | `48` | `59` | `11` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5516` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18422/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18422/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

File: source/content/advanced-redirects.md (hunk 2 above)
Comment (alexfornuto) [reply source/content/advanced-redirects.md]: Thanks, fixed in https://github.com/pantheon-systems/documentation/pull/5516/commits/8bbec23b958aacb434a1a517ec299e28c9a557af

File: source/content/advanced-redirects.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+title: Advanced Redirects and Restrictions
+description: Configure custom redirect logic for specific scenarios
+tags: [redirects, variables, dns]
+categories: [go-live,develop]
+reviewed: "2020-02-12"
+---
+
+Basic domain and HTTPS redirection can be handled by the [Primary Domain]() feature. PHP redicts can be used if these configurations are not an option, or when you need specific redirect logic. Depending on the needs of the site, you may only need one, some, or none of the following.
+
+Redirect logic should be added to `wp-config.php` for WordPress sites, and `settings.php` for Drupal sites. Note that the platform-set primary domain will redirect all requests, not just the root domain.
```
Comment (rachelwhitton) [inline source/content/advanced-redirects.md]: Consider using a warning or note for the second sentence here - worth a stronger call out imo

File: source/content/advanced-redirects.md [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+## Restrict File Access Based on IP
+
+If you want to restrict access to files based on the source IP address of the request, you can do so with PHP.
+
+<TabList>
+
+<Tab title="WordPress" active={true} id="restrict-wp">
+
+The following example restricts access to `/wp-admin/` and `/wp-login.php` based on the IP addresses listed in the `trusted_ips` array:
```
Comment (rachelwhitton) [inline source/content/advanced-redirects.md]: Replace `trusted_ips` with `$trusted_ips`

File: source/content/advanced-redirects.md [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+        foreach($ips as $ip) {
+          if(FALSE !== strpos($check_ip, $ip)) {
+            return true;
+          }
+        }
+    }
+    return false;
+  }
+function is_from_trusted_ip() {
+   $trusted_ips = array('192.0.2.38','198.51.100.12','208.0.113.159','2001:DB8:1C93');
```
Comment (rachelwhitton) [inline source/content/advanced-redirects.md]: Insert a comment above this line that instructs the reader to replace IPs

Comment (rachelwhitton) [reply source/content/advanced-redirects.md]: Same feedback for drupal snippet

File: source/content/advanced-redirects.md (hunk 4 above)
Comment (rachelwhitton) [reply source/content/advanced-redirects.md]: Same feedback for drupal snippet

File: source/partials/primary-domain.md [hunk 6]
```diff
@@ -1,5 +1,11 @@
 ### Set a Primary Domain via the Dashboard
 
+<Alert type="danger" title="Warning">
+
+With a Primary Domain set at the platform level, all other domains will be pointed to your Primary domain _at the root level_. If you want to redirect secondary domains to specific pages on your site (for example, `olddomain.com` to `newdomain.com/old-landing-page`), do not set a Primary Domain. Instead use [PHP redirects](/redirects/#redirect-with-php).
```
Comment (rachelwhitton) [inline source/partials/primary-domain.md]: Suggest applying a similar warning to the new advanced docs page at the top, which says if they're going to do advanced redirect scenarios the primary domain feature needs to be set to none.

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5516 | Difference |
| --- | --- | --- | --- |
| Performance Score | `80` | `49` | `-31` |
| Best Practice Score | `93` | `86` | `-7` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `98` | `96` | `-2` |

Lighthouse Audit Reports:

- [`pr-5516` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18426/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18426/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

File: source/partials/primary-domain.md (hunk 6 above)
Comment (alexfornuto) [reply source/partials/primary-domain.md]: Added in https://github.com/pantheon-systems/documentation/pull/5516/commits/a7e4c20988a819b75567c926d7678d71063e686a

File: source/content/advanced-redirects.md (hunk 5 above)
Comment (alexfornuto) [reply source/content/advanced-redirects.md]: Comments added in https://github.com/pantheon-systems/documentation/pull/5516/commits/a7e4c20988a819b75567c926d7678d71063e686a

File: source/content/advanced-redirects.md (hunk 4 above)
Comment (alexfornuto) [reply source/content/advanced-redirects.md]: Value corrected in https://github.com/pantheon-systems/documentation/pull/5516/commits/a7e4c20988a819b75567c926d7678d71063e686a

File: source/content/advanced-redirects.md (hunk 3 above)
Comment (alexfornuto) [reply source/content/advanced-redirects.md]: Noted: https://github.com/pantheon-systems/documentation/pull/5516/commits/a7e4c20988a819b75567c926d7678d71063e686a

**Comment by alexfornuto** [comment]
Thanks @rachelwhitton! I added your example of Primary Domain removal verification [here](https://github.com/pantheon-systems/documentation/pull/5516/files#diff-11c3a40ae3025a1506796f5784b72cd7R13-R35)

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5516 | Difference |
| --- | --- | --- | --- |
| Performance Score | `61` | `49` | `-12` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5516` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18430/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18430/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

File: source/data/landings.yaml [hunk 7]
```diff
@@ -413,8 +415,6 @@
               url: '/hosts-file/'
             - text: "HTTPS on Pantheon's Global CDN"
               url: '/https/'
-            - text: 'Configure Redirects'
-              url: '/redirects/'
```
Comment (alexfornuto) [inline source/data/landings.yaml]: FYI to whoever copy-reviews this: Removed because this page was listed twice on the same landing, right next to itself.

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5516 | Difference |
| --- | --- | --- | --- |
| Performance Score | `56.00000000000001` | `59` | `3` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5516` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18434/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18434/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5516 | Difference |
| --- | --- | --- | --- |
| Performance Score | `49` | `46` | `-3` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5516` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18471/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18471/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5516 | Difference |
| --- | --- | --- | --- |
| Performance Score | `55.00000000000001` | `46` | `-9` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5516` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18474/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18474/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Pull request merged**